Several additions, fixes and changes

erty80s opened the pull request:
-Payloads (Nuclear bombs and Anti-matter charges) can be detonated anywhere
-Anti matter charge added to the covenant corvette
-Both payloads have distinct detonations
-Payloads can be loaded into umbilicals
-Payloads and other pullable stuff such as crates can be loaded onto vehicles
-Vehicles can now be loaded onto air vehicles such as pelicans and spirits
-Fixed small issue where pullable objects wouldn't go upstairs when pulled
-Mechs can now go upstairs (but not down stairs)

erty80s committed:
Several additions, fixes and changes

-Payloads (Nuclear bombs and Anti-matter charges) can be detonated anywhere
-Anti matter charge added to the covenant corvette
-Both payloads have distinct detonations
-Payloads can be loaded into umbilicals
-Payloads and other pullable stuff such as crates can be loaded onto vehicles
-Vehicles can now be loaded onto air vehicles such as pelicans and spirits
-Fixed small issure where pullable objects wouldn't go upstairs with the mob
-Mechs can now go upstairs (but not down stairs)

erty80s committed:
Small changes

-Added variations for payloads: ones that explode within an specific areas and other that explode anywhere
-Made the nuclear bomb slightly more deadly
-Replaced the payloads with explode anywhere variations

bloxgate commented on code/modules/halo/misc/payload.dm:
```diff
@@ -7,6 +7,9 @@
 	anchored = 0
 	density = 1
 	var/explodetype = /datum/nuclearexplosion
+	var/explodetype2 = /datum/antiexplosion
```
Just a suggestion, but maybe you could make the antiexplosion a subtype of nuclearexplosion so that you can just change this single variable in that type of bomb instead of adding in snowflake vars for each type of WMD we use?

X0-11 replied:
I'm suprised it was done this way, the original system was meant to take an explosion datum to ensure this didn't happen. Please do what Blox suggested.

erty80s replied:
I fixed my spaghetti code, which in retrospect looked really bad thanks for pointing that out. It should be ok now.

erty80s committed:
Small code fixes

-Added more generalized datums for explosions
-Removed heating for nuclear bomb due to not working correctly

erty80s committed:
Some code clean-up

-General clean-up nothing too big

erty80s commented:
I fixed my spaghetti code, which in retrospect looked really bad thanks for pointing that out. It should be ok now

erty80s commented:
I fixed my spaghetti code, which in retrospect looked really bad thanks for pointing that out. It should be ok now

BDpuffy420 commented:
PR needs a changelog and there are conflicting files

erty80s commented:
Actually James I might hold on on fixing this for maybe 2 days because I'm adding more new stuff. So this pr might stay open a lil bit longer.

BDpuffy420 commented:
Righto, adding  wip tag

erty80s committed:
Orbital bombardment and other stuff

-Orbital bombardment reinstated
-Mac rounds no longer hit random z-levels and only hit the upper-z levels
-New damaged icons for exoplanets

erty80s committed:
Kig-Yar ranger suits

-Adds kig-yar ranger suits

erty80s committed:
Adds changelog

-Adds the changelog
-Reverts changes to all maps

erty80s commented:
I reverted the changes to all maps, all I did was replace the payload in oni base and add a small room with the anti matter charge. I'll leave that job to mappers or someone else because I don't know why it has conflicts and how to fix them. Also, the kig yar ranger suit should be added to the corvette along with masks and oxygen tanks and yes the pr should be ready to merge in theory (hope there's no more issues with it)

BDpuffy420 commented:
PR is failing

erty80s commented:
Whats wrong wit it?

EDIT: I might need to compile hold on

erty80s committed:
Merge branch 'alpha' of https://github.com/HaloSpaceStation/HaloSpaceStation13 into alpha

erty80s committed:
fixing for travis checks

just making sure it doesnt fail any more checks

erty80s committed:
fixing for travis checks 2

just making sure it doesnt fail any more checks

erty80s committed:
Some code changes

modifying some code so it passes the travis fixes

erty80s committed:
Code clean-up

-Deleted free_explode payloads and made it a variable instead
-Edited the changlog a bit hoping it passes travis

erty80s committed:
Small changelog edit

erty80s committed:
Another edit hoping it pleases travis

erty80s commented:
Ok now that it passed all travis checks it is ready to be merged, take a good look at some of the values for the kig yar armor and the explosions made by the payloads and mac. If there's any issues regarding that ill change them.

NOTE: The payloads that explode freely and the kig-yar suit are NOT in the game yet. Once this pr gets merged I'll add them in another one just to avoid more conflicts with this one
